src/licensedcode/match_unknown.py

```diff
@@ -0,0 +1,77 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# ScanCode is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/nexB/scancode-toolkit for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+
+from licensedcode import tokenize
+"""
+Matching strategy for unknown matching using ngrams.
+"""
+
+# Set to False to enable debug tracing
+TRACE = False
+
+if TRACE:
+    import logging
+    import sys
+
+    logger = logging.getLogger(__name__)
+
+    def logger_debug(*args):
+        return logger.debug(' '.join(isinstance(a, str) and a or repr(a) for a in args))
+
+    logging.basicConfig(stream=sys.stdout)
+    logger.setLevel(logging.DEBUG)
+
+else:
+
+    def logger_debug(*args):
+        pass
+
+MATCH_UNKNOWN = '6-unknown'
+
+
+def add_ngrams(automaton, tids, rule_length, unknown_ngram_length=7):
```

Comment:
What is `rule_length` doing here?

Reply:
We need to check if our rule_length is less than our n-gram length right ?